base/managers.jl

```diff
@@ -209,6 +209,31 @@ end
 immutable DefaultClusterManager <: ClusterManager
 end
 
+type Sem
```

Comment:
It seems weird to qualify that the semaphore is non-reentrant. How about just calling it `Semaphore` and if/when we have a reentrant version, calling that `ReentrantSemaphore`.